src/datepicker/themes/bs/bs-days-calendar-view.component.ts

```diff
@@ -90,22 +90,24 @@ export class BsDaysCalendarViewComponent  {
   }
 
   selectWeek(week: WeekViewModel): void {
-    if (!this._config.selectWeek) {
+    if (!this._config.selectWeek && !this._config.selectWeekDateRange) {
       return;
     }
 
-    if (week.days
-      && week.days[0]
-      && !week.days[0].isDisabled
-      && this._config.selectFromOtherMonth) {
-
-      this.onSelect.emit(week.days[0]);
-
+    if (week.days.length === 0) {
       return;
     }
 
-    if (week.days.length === 0) {
-      return;
+    if (this._config.selectWeek) {
+      if (week.days
```

Comment:
Could you please merge these conditions?

Reply:
:heavy_check_mark: